### DISCUSSION TITLE
Godot audiostreamrandomizer looping

### DISCUSSION THREAD

**Discussion by DorusvdLinden** [missionpinball/missionpinball#149] [Q&A] [answered]
Dear all,

I love the new setup, but i have one issue: when I use the loops setting in MPF 0.80.0 on an AudioStreamRandomizer The godot script stops. Is there a way to continuously loop multiple sounds in a random order?

edit: 
In my previous iteration I was looping multiple sounds in attract mode randomly, to create a dynamic background.

Now in Godot, the individual wav files can loop, but the audiostreamrandomiser stops after 1 sound played.

Using loops: -1, 3 or true gives an error in the MPF.Media.Sounds code (GMCChannel.gd) in mpf-gmc/scripts/) (I also posted this error at: https://github.com/missionpinball/mpf-gmc/issues/15)
![Screenshot 2024-09-14 153948](https://github.com/user-attachments/assets/5136a617-ef54-416c-9c71-c9852163bd59)
![Screenshot 2024-09-14 153751](https://github.com/user-attachments/assets/0193e5fb-5f6c-4240-87bc-2272473d2b23)

code:
```
sound_player:
  mode_attract_started:
    SlowMusic:
      bus: music
      loops: 3
```

without the loops it runs fine.
Could be my mistake.
Kind regards
Dorus

**Comment by avanwinkle** [comment, marked as answer]
Great find, see https://github.com/missionpinball/mpf-gmc/pull/16 for a potential feature to support loops!

**Reply by DorusvdLinden** [reply]
Thank you, I will test this weekend!

**Reply by DorusvdLinden** [reply]
Testing brought up a new error in the code when using loops:-1, I have updated the issue. The fixed number of loops works now.
